src/components/EncounterDetails/EncounterDetails.jsx

```diff
@@ -1,5 +1,165 @@
+            <h3>Legendary Action: {action.name}</h3>
+            <p>{action.desc}</p>
+          </div>
+        ));
+
+        const specialAbilitiesArray = monster.special_abilities.map((ability) => (
+          <div className='black foe-special-abilities' key={ability.name}>
+            <h3>Special Ability: {ability.name}</h3>
+            <p>{ability.desc}</p>
+          </div>
+        ));
+
+        return (
+          <details key={monster.name}>
+            <summary>{monster.name}</summary>
+            <section className='black foe-physical'>
+              <h3>Size:</h3>
+              <p>{monster.size}</p>
+              <h3>Armor Class:</h3>
+              <p>{monster.armor_class.value || 0}</p>
+              <h3>Hit Points:</h3>
+              <p>{monster.hit_points || 0}</p>
+            </section>
+            <section className='black foe-stats'>
+              <div className='attributes'>
+                <h3>Strength</h3>
+                <p>{monster.strength || 0}</p>
+              </div>
+              <div className='attributes'>
+                <h3>Dexterity</h3>
+                <p>{monster.dexterity || 0}</p>
+              </div>
+              <div className='attributes'>
+                <h3>Constitution</h3>
+                <p>{monster.constitution || 0}</p>
+              </div>
+              <div className='attributes'>
+                <h3>Intelligence</h3>
+                <p>{monster.intelligence || 0}</p>
+              </div>
+              <div className='attributes'>
+                <h3>Wisdom</h3>
+                <p>{monster.wisdom || 0}</p>
+              </div>
+              <div className='attributes'>
+                <h3>Charisma</h3>
+                <p>{monster.charisma || 0}</p>
+              </div>
+            </section>
+            <section className='black foe-speeds'>
+              <div className='attributes'>
+                <h3>Walk Speed:</h3>
+                <p>{monster.speed.walk || "0 ft."}</p>
+              </div>
+              <div className='attributes'>
+                <h3>Swim Speed:</h3>
+                <p>{monster.speed.swim || "0 ft."}</p>
+              </div>
+              <div className='attributes'>
+                <h3>Fly Speed:</h3>
+                <p>{monster.speed.fly || "0 ft."}</p>
+              </div>
+              <div className='attributes'>
+                <h3>Burrow Speed:</h3>
+                <p>{monster.speed.burrow || "0 ft."}</p>
+              </div>
+            </section>
+            <section className='black damage-mods'>
+              <div>
+                <h3>Vulnerabilities:</h3>
+                <p>{monster.damage_vulnerabilities}</p>
+              </div>
+              <div>
+                <h3>Resistances:</h3>
+                <p>{monster.damage_resistances}</p>
+              </div>
+              <div>
+                <h3>Immunities:</h3>
+                <p>{monster.damage_immunities}</p>
+              </div>
```

Comment:
@tenthwalker In every other category, you've added a conditional to show "0 ft" or "0" or "none", etc, but not in these three. Was that oversight? Or did you leave them blank intentionally? Ideally, we'd stay consistent unless there's a specific reason to deviate here.

Reply:
I couldn't get them to render the value on the other side of the `||` operator, so I took the conditional out on those three with the intent to ask you if you had ideas and then totally forgot to 🤦🏼 Thanks for catching this! I tried it with both an array containing the string of none, because the actual data is in an array, as well as just a string of "none" which also did not render.

Reply:
I think the issue is that each of those returns an array. So, it has to be mapped over. If you have more than one value in the array right now, it's going to mush them together, so we probably need a join in there, too? (e.g. If I add "poison" to a dragon's immunities, it'll print in the box as "firepoison")

Reply:
Pushed the fix for this, I was able to have it map and render a `<p>` element for each value or "none" if the array was length === 0

Reply:
That works for me! Nicely done! ⭐